Improve error and documentation on the needed link between router and service (#10262)
traefik/traefik: docs/content/providers/docker.md

    --providers.docker.swarmMode=true
    ```

>     Attach labels to a single service (not to containers) while in Swarm mode (in your docker compose file)
>     When there is only one service and the router does not specify any service,
>     then that service is automatically assigned to the router.

Reviewer: ```suggestion
    Attach labels to a single service (not containers) while in Swarm mode (in your Docker compose file).
    If there is only one service, and the router does not specify a service,
    then that service is automatically assigned to the router.
```

        deploy:
          labels:
            - traefik.http.routers.my-container.rule=Host(`example.com`)

>             - traefik.http.routers.my-container.service=my-container-service

Reviewer 2: ```suggestion
```

Author: @ldez @bluepuma77 @mmatur With ceed9c579, I tried to improve the wording and keep current "magic" example. I also added a complete example in service definition section.

Author: Implemented with https://github.com/traefik/traefik/commit/ceed9c579696f26fe6fc54059619775671e27a69
